src/modules/formBuilder/components/formTree/useUpdateFormStructure.tsx

```diff
@@ -12,10 +18,11 @@ import {
   getPathContext,
   insertItemToDefinition,
   removeItemFromDefinition,
+  validateDeletion,
 } from '@/modules/formBuilder/formBuilderUtil';
 import { FormDefinitionJson, FormItem, ItemType } from '@/types/gqlTypes';
 
-export default function useReorderItem(
+export default function useUpdateFormStructure(
```

Comment:
trying to comment on line 34 `const itemIdMap = useMemo(() => getItemIdMap(values.item), [values]);` -- does this need to be re-calculated here? It looks like you added it to the context, so you could consume it from that, or better yet just pass it as an argument to this hook: `useUpdateFormStructure(control, itemMap, itemId)`

Reply:
Hm, this is confusing, let me think about this a bit more. The problem is that there are two different maps with too-similar names, `itemMap` and `itemIdMap`. I think we should rename them for clarity, and probably also improve where they are generated and how they are passed around for consistency and performance.
- `itemMap`:
  - Maps `linkId => FormItem`
  - Predates the form builder, is used many places across form related code
  - For the Form Builder, it's currently created inside of `FormTree` and passed via `FormTreeContext` so that it doesn't get re-generated by every `FormTreeLabel`, which is what was happening before [this commit](https://github.com/greenriver/hmis-frontend/pull/821/commits/3bef69c92f20a396eee618282e30cd5cb51545eb).
  - It's used by the `FormTreeLabel` in order to grab the item from the form definition instead of from RichTreeView's "publicAPI", which is what was happening before [this commit](https://github.com/greenriver/hmis-frontend/pull/821/commits/063f06aa47f006a364da18c901b1596b4b7e2f7e#diff-0531236b7fea21dc33f861077b3ca9822708adeb428d830d25d7c72ee92dd46eL51-L52).
- `itemIdMap`
  - Maps `linkId => path` where path is e.g. `item.0.item.1`
  - New with the RHF/form builder
  - Only used here in the `useUpdateFormStructure` hook in order to get the RHF paths for manipulating the location of items within the tree.

Whew thanks for the opportunity to parse out that confusion in writing! I think my proposed changes are:
- rename `itemIdMap`, maybe `itemPathMap`? `itemRhfPathMap`? `rhfPathMap`?
- move `itemIdMap`'s generation into `FormTree` and pass to `FormTreeContext` alongside `itemMap`, that way it isn't regenerated per `FormTreeLabel`.

Reply:
Ok, I've attempted to clarify a bit in this commit: https://github.com/greenriver/hmis-frontend/pull/821/commits/78c2e729d494aba64034ed775c23c422f3f5a07b

Reply:
Oh sorry, totally missed that getItemIdMap and getItemMap were different! Sounds good